examples/rest-stickies-csv/app.js

```diff
@@ -32,964 +38,273 @@ app.engine(
-let tagData3;
-let tagData4;
-
-// Clean Tag IDs from response
-let tagId;
-let tagId2;
-let tagId3;
-let tagId4;
-
-// Declare variables for connector endpoint ids
-let endpoint1Id;
-let endpoint2Id;
-let endpoint3Id;
-let endpoint4Id;
-let endpoint5Id;
-let endpoint6Id;
-let endpoint7Id;
-let endpoint8Id;
-let endpoint9Id;
-let endpoint10Id;
-let endpoint11Id;
-let endpoint12Id;
-
 // <-------- ROUTES -------->
 
 // ROUTE (GET): Retrieve access_token from OAuth redirect
-app.get("/authorized", (req, res) => {
-  let parsedUrl = req.url;
-  let urlValues = parsedUrl.split("?");
-  console.log(urlValues[1]);
-  oauthAccessToken = urlValues[1];
+app.get("/authorized", async (req, res) => {
+  try {
+    await miro.handleAuthorizationCodeRequest(USER_ID, req);
+  } catch (e) {
+    console.error(e);
+  }
 
   res.render("authorizeApp");
 });
 
 // ROUTE (POST): Trigger the authorization flow
-app.post("/", function (req, res) {
-  if (oauthAccessToken) {
+app.post("/", async function (req, res) {
+  if (await miro.isAuthorized(USER_ID)) {
     res.render("authorizeApp");
-  } else {
-    res.redirect(
-      "https://miro.com/oauth/authorize?response_type=code&client_id=" +
-        process.env.clientID +
-        "&redirect_uri=" +
-        process.env.redirectURL
-    );
+    return;
   }
+
+  res.redirect(miro.getAuthUrl());
 });
 
 // ROUTE(GET): VIEW UPLOAD .CSV OPTION
-app.get("/upload-csv", (req, res) => {
-  if (oauthAccessToken) {
+app.get("/upload-csv", async (req, res) => {
+  if (await miro.isAuthorized(USER_ID)) {
     res.render("uploadCSV");
   } else {
-    res.redirect(
-      "https://miro.com/oauth/authorize?response_type=code&client_id=" +
-        process.env.clientID +
-        "&redirect_uri=" +
-        process.env.redirectURL
-    );
+    res.redirect(miro.getAuthUrl());
   }
 });
 
 // ROUTE(POST): UPLOAD .CSV FILE
-app.post("/upload-csv", upload.single("csv"), function (req, res) {
-  //console.log("upload triggered, here is req :" + req)
-  const fileRows = [];
-
-  // open uploaded file
-  csv
-    .parseFile(req.file.path)
-    .on("data", function (data) {
-      fileRows.push(data); // push each row
-    })
-    .on("end", function () {
-      console.log(fileRows);
-      fs.unlinkSync(req.file.path); // remove temp file
-      fileRows.shift(); // remove csv headers (start with actual content)
-    });
+app.post("/upload-csv", upload.single("csv"), async function (req, res) {
+  if (!(await miro.isAuthorized(USER_ID))) {
+    res.redirect(miro.getAuthUrl());
+  }
+  let fileRows = ["No file uploaded"];
+
+  if (req.file) {
+    fileRows = [];
+    // open uploaded file
+    csv
+      .parseFile(req.file.path)
+      .on("data", function (data) {
+        fileRows.push(data); // push each row
+      })
+      .on("end", function () {
+        fs.unlinkSync(req.file.path); // remove temp file
+        fileRows.shift(); // remove csv headers (start with actual content)
+      });
+    console.log(fileRows);
```

Comment:
nitpick: not sure if this is important but it will always render empty array (if it's put back in the "end" handler it will log the parsed rows